RadixWallet/Features/TransactionReviewFeature/SubmitTransaction/SubmitTransaction.swift

```diff
@@ -117,93 +122,104 @@ public struct SubmitTransaction: Sendable, FeatureReducer {
+				do {
+					try await submitTXClient.hasTXBeenCommittedSuccessfully(txID)
+					await send(.internal(.statusUpdate(.committedSuccessfully)))
+				} catch let error as TXFailureStatus {
+					// Error is always TXFailureStatus, just that it is erased to generic Error
+					switch error {
+					case .permanentlyRejected:
+						await send(.internal(.statusUpdate(.permanentlyRejected)))
+					case let .temporarilyRejected(epoch):
+						await send(.internal(.statusUpdate(
+							.temporarilyRejected(remainingProcessingTime: Int(endEpoch - epoch.rawValue) * Self.epochDurationInMinutes)
+						)))
+					case .failed:
+						await send(.internal(.statusUpdate(.failed)))
 					}
-					await send(.internal(.statusUpdate(update.result)))
 				}
-			} catch: { error, send in
-				loggerGlobal.error("Failed to receive TX status update, error \(error)")
-				await send(.internal(.statusUpdate(.failure(.failedToGetTransactionStatus(txID: txID, error: .init(pollAttempts: pollStrategy.maxPollTries))))))
 			}
-
-		case let .statusUpdate(update):
-			switch update {
-			case let .success(status):
-				let stateStatus = status.stateStatus
-				state.status = stateStatus
-				if stateStatus.isCompletedSuccessfully {
-					return .send(.delegate(.committedSuccessfully(state.notarizedTX.txID)))
-				} else if stateStatus.isSubmitted {
-					if !state.hasDelegatedThatTXHasBeenSubmitted {
-						defer { state.hasDelegatedThatTXHasBeenSubmitted = true }
-						return .send(.delegate(.submittedButNotCompleted(state.notarizedTX.txID)))
-					}
-				}
-				return .none
-			case .failure:
-				/// Need to show failure
-				state.status = .failedToGetStatus
+			.cancellable(id: CancellableId.transactionStatus, cancelInFlight: true)
+			.merge(with: .send(.delegate(.submittedButNotCompleted(state.notarizedTX.txID))))
+
+		case let .statusUpdate(status):
+			state.status = status
+			if status == .committedSuccessfully {
+				return transactionCommittedSuccesfully(state)
+			} else {
 				return .none
 			}
 		}
 	}
-}
-
-extension GatewayAPI.TransactionStatus {
-	var stateStatus: SubmitTransaction.State.TXStatus {
-		switch self {
-		case .committedFailure: .committedFailure
-		case .committedSuccess: .committedSuccessfully
-		case .pending: .submittedPending
-		case .rejected: .rejected
-		case .unknown: .submittedUnknown
-		}
-	}
-}
 
-extension SubmitTransaction.State.TXStatus {
-	var isComplete: Bool {
-		isCompletedWithFailure || isCompletedSuccessfully
-	}
-
-	var isSubmitted: Bool {
-		switch self {
-		case .failedToGetStatus, .rejected, .committedFailure, .submittedUnknown, .submittedPending, .committedSuccessfully: true
-		case .submitting, .notYetSubmitted: false
-		}
-	}
+	private func transactionCommittedSuccesfully(_ state: State) -> Effect<Action> {
+		// TODO: Could probably be moved in other place. TransactionClient? AccountPortfolio?
+		Task.detached { [intent = state.notarizedTX.intent] in
```

Comment:
hmm yes this is wrong, cannot be here in Reducer, even though `detached` I THINK this will be cancelled when the reducer feature is dismissed? did you check? can probably be put into a client (again)?

Reply:
I did check, `detached` task is not cancelled, since it is a top-level task. Moved to AccountPortfoliosClient.